locust/test/test_main.py

```diff
@@ -280,46 +412,104 @@ class TestUser(User):
                 @task
                 def my_task(self):
                     print("running my_task()")
-        """
+            """
             )
         ) as file_path:
-            proc = subprocess.Popen(["locust", "-f", file_path, "--autostart"], stdout=PIPE, stderr=PIPE, text=True)
-            gevent.sleep(1)
-            stdout, stderr = proc.communicate()
-            self.assertIn("parameter need to be float and value between. 0 < percentile < 1 Eg 0.95", stderr)
-            self.assertEqual(1, proc.returncode)
+            proc = subprocess.Popen(
+                ["locust", "-f", file_path, "--autostart"],
+                stdout=PIPE,
+                stderr=PIPE,
+                text=True,
+                bufsize=1,  # Line-buffered output
+            )
+
+            try:
+                # Continuously read stderr for real-time updates
+                stderr_output = []
+
+                def check_for_error_message():
```

Comment:
for this case it is not worth polling, because the expected result is for the process to exit completely on its own, and the polling would make a failure harder to read.